lib/hardware/modules/iob_add2/iob_add2.py

```diff
@@ -3,7 +3,79 @@ def setup(py_params_dict):
         "original_name": "iob_add2",
         "name": "iob_add2",
         "version": "0.1",
-        "generate_hw": False,
+        "confs": [
+            {
+                "name": "W",
+                "type": "P",
+                "val": "21",
+                "min": "NA",
+                "max": "NA",
+                "descr": "Data bus width",
+            },
+        ],
+        "ports": [
+            {
+                "name": "in1_i",
+                "descr": "Input port",
+                "signals": [
+                    {
+                        "name": "in1",
+                        "width": "W",
+                        "direction": "input",
+                    },
+                ],
+            },
+            {
+                "name": "in2_i",
+                "descr": "Input port",
+                "signals": [
+                    {
+                        "name": "in2",
+                        "width": "W",
+                        "direction": "input",
+                    },
+                ],
+            },
+            {
+                "name": "sum_o",
+                "descr": "Output port",
+                "signals": [
+                    {
+                        "name": "sum",
+                        "width": "W",
+                        "direction": "output",
+                    },
+                ],
+            },
+            {
+                "name": "carry_o",
+                "descr": "Output port",
+                "signals": [
+                    {
+                        "name": "carry",
+                        "width": 1,
+                        "direction": "output",
+                    },
+                ],
+            },
+        ],
+        "wires": [
+            {
+                "name": "sum_int",
+                "descr": "sum wire",
+                "signals": [
+                    {"name": "sum_int", "width": "W+1"},
+                ],
+            },
+        ],
+        "snippets": [
+            {
+                "verilog_code": f"""
+    assign sum_o   = in1_i + in2_i;
+    assign carry_o = sum_o[W];
```

Comment:
It looks like the [original implementation](https://github.com/IObundle/iob-soc/pull/875/files#diff-c0667e001eea98d153a3d0c2c187ae68f16a235dc89e4bffd43f897cfb353fa7L11-L13) of this module was wrong.
We should use the internal `sum_int` wire to store the result of the sum, including the carry bit. 
```suggestion
    assign sum_int = in1_i + in2_i;
    assign sum_o = sum_int[W-1:0];
    assign carry_o = sum_int[W];
```